src/vm/eventpipebuffer.cpp

```diff
@@ -144,65 +149,49 @@ EventPipeEventInstance *EventPipeBuffer::GetNext(EventPipeEventInstance *pEvent,
     CONTRACTL_END;
 
     EventPipeEventInstance *pNextInstance = NULL;
-    // If input is NULL, return the first event if there is one.
-    if (pEvent == NULL)
-    {
-        // If this buffer contains an event, select it.
-        BYTE *pFirstAlignedInstance = GetNextAlignedAddress(m_pBuffer);
-        if (m_pCurrent > pFirstAlignedInstance)
-        {
-            pNextInstance = (EventPipeEventInstance *)pFirstAlignedInstance;
-        }
-        else
-        {
-            return NULL;
-        }
-    }
-    else
+
+    // If m_pCurrentReadEvent is NULL we've reached the end of the events
+    if (m_pCurrentReadEvent != NULL)
     {
         // Confirm that pEvent is within the used range of the buffer.
-        if (((BYTE *)pEvent < m_pBuffer) || ((BYTE *)pEvent >= m_pCurrent))
+        if (((BYTE*)m_pCurrentReadEvent < m_pBuffer) || ((BYTE*)m_pCurrentReadEvent >= m_pCurrent))
```

Comment:
Is this check meant to be outside this if-block?

Reply:
I think its correct where it is, was there anything specific that looked bad? (Aside from that AV below which is definately a bug : )

Reply:
The comment was about the AV below.